### PR TITLE
1563 GitHub best practices

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Improving GitHub source

## How
1. Including all streams to configured catalog for standard test
2. Updating standard test
3. Installing tap-github from fork instead PyPI for handling rate limits

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

File: airbyte-integrations/connectors/source-github-singer/setup.py [hunk 1]
```diff
@@ -33,7 +33,13 @@
     package_data={"": ["*.json"]},
     # two sets of dependencies: 1) for main 2) for standard test deps. 2 does not have all of the dependencies of 1, which is we cannot use install_requires.
     extras_require={
-        "main": ["tap-github==1.9.0", "requests", "airbyte-protocol", "base-singer", "base-python"],
+        "main": [
+            "tap-github @ https://github.com/vitaliizazmic/tap-github/tarball/master#egg=tap-github-1.9.1",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-github-singer/setup.py:37]: is it possible to use this fork instead: https://github.com/airbytehq/tap-github ? it is identical to yours, but having it under the Airbyte organization/brand is good for external visibility.

**Review by sherifnada — COMMENTED** [review]
Looks like standard tests are failing because we are trying to sync all streams for both repos in the config file, but the `teams` resource is not available for the private personal repo in the test case. WDYT about doing the following: 
1. remove the private repo from the config, and 
2. add a custom integration test case which attempts to read one stream from the  private repo just to make sure we can access it

File: airbyte-integrations/connectors/source-github-singer/setup.py (hunk 1 above)
Comment (vitaliizazmic) [reply airbyte-integrations/connectors/source-github-singer/setup.py]: Yes, I've replaced it.

**Comment by vitaliizazmic** [comment]
> Looks like standard tests are failing because we are trying to sync all streams for both repos in the config file, but the `teams` resource is not available for the private personal repo in the test case. WDYT about doing the following:
> 
> 1. remove the private repo from the config, and
> 2. add a custom integration test case which attempts to read one stream from the  private repo just to make sure we can access it

1. There are several cases, when syncing Team stream fails: 

- Private repository. In this case Github API returns 404 error with message "Not Found" 
- Accessing org teams without admin permissions. In this case Github API returns 403 error with message "Must have admin rights to Repository." 

2. Yes, we can do separate test case for private repositories. Commits stream is using by default for checking configs.

**Pull request merged**